### PR TITLE
#162948876 Delete an article

### PR DESCRIPTION
**Pull request by tesh254**
#### What does this Pull Request do?

The PR contains the component that enables a user to delete an article

#### Description of tasks to be accomplished?

- Integrate Redux into the functionality.
- Create the Delete article component
- Warn user on delete attempt

#### How can this be manually tested?

* Pull the branch `git fetch origin ft-liking-162948880:ft-liking-162948880`
* Checkout to the branch `git checkout ft-liking-162948880`
* Install dependencies `npm install`
* Run the server `npm start`

#### Screenshots

screenshot 2019-02-28 at 6 24 32 pm

screenshot 2019-02-28 at 6 26 09 pm

#### Related Pivotal Tracker story?

[#162948876](https://www.pivotaltracker.com/story/show/162948876)

### CODE REVIEW

File: src/components/containers/articles/DeleteArticle.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      );
+    }
+  };
+
+  deleteAction = () => {
+    this.handleConfirm();
+    const article = this.props.article;
+    const slug = article.article.slug;
+    this.props.deleteArticle(slug);
+    console.log(slug);
```
[the remark this answers is not in the document]
Comment (Deekerubo) [reply src/components/containers/articles/DeleteArticle.js]: Remove this console.log statement, please!

File: src/components/containers/articles/OneArticle.js [hunk 2]
```diff
@@ -71,20 +72,30 @@ export class GetOneArticle extends Component {
...
+                        </h4>
+                        {!localStorage.getItem("token") ||
+                        localStorage.getItem("token") === undefined ? (
                           <div />
-                          ) : localStorage.getItem("username") === article.author.username ? (
-                            <Link to={`/${this.props.props.match.params.slug}/edit`} id="the-btn">
-                              <button className="ui black medium button follow-button edit-button">Edit</button>
+                        ) : localStorage.getItem("username") ===
+                          article.author.username ? (
+                          <div>
```
[the remark this answers is not in the document]
Comment (Deekerubo) [reply src/components/containers/articles/OneArticle.js]: Format your code to align your indentation.

**Comment by danuluma** [comment]
The confirmation message seems to be too long

**Comment by boltonotieno** [comment]
Your implementation is good, but the confirmation message for delete is a bit off. Would you change it to something short

**Comment by danuluma** [comment]
Should be just 2-3 words long....something like just "Confirm delete" or "Are you sure?"

File: src/assets/css/index.scss [hunk 3]
```diff
@@ -396,3 +396,7 @@ img.ui.avatar.image {
 .rating-bar {
   margin: 1em 0 2em 0 !important;
 }
+.action:hover {
+  cursor: pointer;
+  color: black;
+}
```
Comment (danuluma) [inline src/assets/css/index.scss]: Please add a newline at the end of this file

File: src/components/containers/articles/DeleteArticle.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (danuluma) [reply src/components/containers/articles/DeleteArticle.js]: Please remove this console.log statement

File: src/components/containers/articles/__test__/DeleteArticle.test.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  mapStateToProps,
+  mapDispatchToProps,
+  RemoveArticle
+} from "../DeleteArticle";
+
+describe("Delete Article", () => {
+    let props;
+    let wrapper;
+    beforeEach(() => {
+        props = {
```
[the remark this answers is not in the document]
Comment (danuluma) [reply src/components/containers/articles/__test__/DeleteArticle.test.js]: Please fix this indentation issue

File: src/components/containers/articles/__test__/DeleteArticle.test.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import {
+  mapStateToProps,
+  mapDispatchToProps,
+  RemoveArticle
+} from "../DeleteArticle";
+
+describe("Delete Article", () => {
+    let props;
+    let wrapper;
+    beforeEach(() => {
```
[the remark this answers is not in the document]
Comment (danuluma) [reply src/components/containers/articles/__test__/DeleteArticle.test.js]: Please fix this indentation issue

File: src/actions/__test__/deleteArticle.test.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  it("delete article", async() => {
+    moxios.wait(() => {
+      const request = moxios.requests.mostRecent();
+      request.respondWith({
+        status: 404
+      });
+    });
+
+    const store = mockStore({});
+    let dispatchedResults = await store.dispatch(deleteArticle("some-article"));
```
[the remark this answers is not in the document]
Comment (Ruiru11) [reply src/actions/__test__/deleteArticle.test.js]: use const here since the value of the variable  does not change

**Review by Ruiru11 — APPROVED** [review]
the code looks good and  the app is running good work

**Review by Deekerubo — APPROVED** [review]

**Review by danuluma — APPROVED** [review]
Works fine

**Review by boltonotieno — DISMISSED** [review]
Update your PR screenshot to the latest Confirm message

**Pull request merged**